bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexServerHandler.java

```diff
@@ -0,0 +1,306 @@
+        int playerCount = 0;
+        int playerActiveCount = 0;
+        Iterator<PlexPlayerHandler> valueIterator = playerHandlers.values().iterator();
+        while (valueIterator.hasNext()) {
+            playerCount++;
+            valueIterator.next().setFoundInSession(false);
+        }
+        if (sessionData != null) {
+            if (sessionData.getSize() > 0) { // Cover condition where nothing is playing
+                for (MediaContainer.MediaType tmpMeta : sessionData.getMediaTypes()) { // Roll through mediaType objects
+                                                                                       // looking for machineID
+                    if (playerHandlers.get(tmpMeta.getPlayer().getMachineIdentifier()) != null) { // if we have a player
+                                                                                                  // configured, update
+                                                                                                  // it
+                        tmpMeta.setArt(plexAPIConnector.getURL(tmpMeta.getArt()));
+                        if (tmpMeta.getType().equals("episode")) {
+                            tmpMeta.setThumb(plexAPIConnector.getURL(tmpMeta.getGrandparentThumb()));
+                            tmpMeta.setTitle(tmpMeta.getGrandparentTitle() + " : " + tmpMeta.getTitle());
+                        } else if (tmpMeta.getType().equals("track")) {
+                            tmpMeta.setThumb(plexAPIConnector.getURL(tmpMeta.getThumb()));
+                            tmpMeta.setTitle(tmpMeta.getGrandparentTitle() + " - " + tmpMeta.getParentTitle() + " - "
+                                    + tmpMeta.getTitle());
+                        } else {
+                            tmpMeta.setThumb(plexAPIConnector.getURL(tmpMeta.getThumb()));
+                        }
+                        playerHandlers.get(tmpMeta.getPlayer().getMachineIdentifier()).refreshSessionData(tmpMeta);
+                        playerActiveCount++;
+                    }
+                }
+            }
+        }
+        updateState(new ChannelUID(getThing().getUID(), PlexBindingConstants.CHANNEL_SERVER_COUNT),
+                new StringType(String.valueOf(playerCount)));
+        updateState(new ChannelUID(getThing().getUID(), PlexBindingConstants.CHANNEL_SERVER_COUNTACTIVE),
+                new StringType(String.valueOf(playerActiveCount)));
+    }
+
+    /**
+     * Refresh all the configured players
+     */
+    private void refreshAllPlayers() {
+        Iterator<PlexPlayerHandler> valueIterator = playerHandlers.values().iterator();
+        while (valueIterator.hasNext()) {
+            valueIterator.next().updateChannels();
+        }
+    }
+
+    /**
+     * This is called to start the refresh job and also to reset that refresh job when a config change is done.
+     */
+    private synchronized void onUpdate() {
+        if (pollingJob == null || pollingJob.isCancelled()) {
```

Comment:
The field can be null at this point, because it isn't inittialized in the constructor. So, you need to check if it is not null before accessing it: Example: if (job != null). But this would result in a compiler warning, as it is a field and could theoretically be modified by another thread exactly between the null check and the access in the next line. To mitigate this, you can store the field to a local variable before checking it:
```
Future<?> localPollingJob = pollingJob;
if (localPollingJob != null) {
    localPollingJob.cancel(true);
}
```

Please check all.

Reply:
I don't see this case. If the first condition is met `pollingJob == null` it doesn't execute the next `pollingJob.isCancelled()`. In the next lines the `pollingJob` is overwritten.

Reply:
Like fwolter said; as pollingJob can be changed outside the scope of this method. A race condition might happen that in between those two checks it changes from non-null to null. I agree that this might never ever happen, but it is a best practice to create a local var for this case.